### DISCUSSION TITLE
Nx Workspace

### DISCUSSION THREAD

**Discussion by dianjuar** [bikecoders/ngx-deploy-npm#76] [Ideas]
Let's discuss new ideas related to Nx Workspaces

**Comment by dianjuar** [comment]
@spaceribs I saw that you commented something on [#73](https://github.com/bikecoders/ngx-deploy-npm/discussions/73#discussioncomment-1447592) that caught my attention:

> unfortunately it is in a private repo, but the command I run in the pull request after the build job looks like this:
> 
> ```bash
> nx affected --target="publish" --tag="pr-${CIRCLE_PR_NUMBER}" --package-version="0.0.0- pr-${CIRCLE_PR_NUMBER}-${CIRCLE_BUILD_NUM}"
> ```

When you install the plugin in a workspace it creates the target `deploy`  but you have `publish` instead.

Do you change that by hand?
That gives the ​idea of creating an option to allow change the name of the target when is doing the install/ng-add.

---

That remembers me something, like a year ago I created an [issue and a PR](https://github.com/nrwl/nx/issues/1830) on Nx to allow running a target only for your libs or projects. The Nx team didn't like the approach that I took. 

One of the solutions they gave me was doing the exact thing that you are doing with the target `publish`, so are you changing `deploy` for `publish` to make sure that it will only run it for your libs?

**Reply by spaceribs** [reply]
That's correct, I manually switched my task to "publish" from "deploy" for a few reasons! You can't roll back a publish, but you can rollback a deploy. My CI allows me to promote and rollback, and I use the different task names to make sure to only run a rollback on applications for a particular environment.

I only publish and automerge the pull request on the production promotion step

**Reply by dianjuar** [reply]
Perfect.
This gives me an idea for a new feature or maybe in the future I could change `deploy` for `publish`

Thank you for sharing

**Comment by dianjuar** [comment]
Since we deprecated angular workspaces, doesn't make sense to have a discussion about the only place this plugin works.

Closing conversation
